src/Manifolds.jl

```diff
@@ -102,6 +106,94 @@ function hat(M::Manifold, x, vⁱ)
     return v
 end
 
+@doc doc"""
+    mean(M,x; weights=1/n*ones(n), stop_tol=10^-7, stop_iter=100 )
```

Comment:
Can you special-case `mean`/`median` for `Euclidean` to forward to the usual functions?

Reply:
Concerning the name, I am against calling it **Frechet**, since thats the global minimizer (see https://en.wikipedia.org/wiki/Fréchet_mean ) which might not be unique on manifolds (due to positive curvature for example) and I am against calling it **Karcher**, since Karcher opposes that (see https://arxiv.org/pdf/1407.2087.pdf), so I prefer to call the function just `mean` and document it as Riemannian Center of Mass.

I can check for the weights and adopt that, surely. And yes, we of course can put that into a separate file and provide fallbacks for Euclidean, since a closed form is known there.

Reply:
> Concerning the name, I am against calling it **Frechet**, since thats the global minimizer (see [https://en.wikipedia.org/wiki/Fréchet_mean](https://en.wikipedia.org/wiki/Fr%C3%A9chet_mean) ) which might not be unique on manifolds

Are you against calling _this_ function a `frechet_mean` (I agree since it's not) or calling _any_ function a `frechet_mean`. If the latter, I don't follow.

If the assumptions behind the COM apply to general manifolds and also give rise to a definition of median and variance, then I agree it makes sense to call the COM `mean`.

Reply:
I haven'T completely understood, how I should model weights here, sorry; can you elaborate on that? I thought just a weight vector can't be that complicated – reading your link, I might have to reconsider.